### DISCUSSION TITLE
Edit Database Driver when editing connection

### DISCUSSION THREAD

**Discussion by rajeshisnepali** [dbeaver/dbeaver#10641] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
![dbever_modify_driver](https://user-images.githubusercontent.com/23610013/101615163-e84d7480-3a35-11eb-99da-a990974dd736.png)
If you see, I have used **mysql5** driver and now I want to update the existing connection as **mysql8** . But I don't think I've getting those options. I had to re-create a new connection with **mysql8** driver.

**Describe the solution you'd like**
There must be an option to edit the driver as well. 

**Describe alternatives you've considered**
Still not. Unfortunately, I've to drop the connection & create a new connection having mysql8 driver.

**Additional context**
Dbeaver Version: Version 7.2.5.202011152110

**Comment by SubOptimal** [comment]
It can be changed from the driver settings.

- got to the driver settings dialog
- there you can download/update the used driver
![Screenshot from 2020-12-10 10-44-49 1](https://user-images.githubusercontent.com/653288/101757702-c4db0600-3ad7-11eb-9674-2ed6595cd7e3.png)

- in the driver settings click on the version column and select the desired driver version

![Screenshot from 2020-12-10 11-02-28 1](https://user-images.githubusercontent.com/653288/101757984-1b484480-3ad8-11eb-8408-922963313463.png)

- with a click on download the driver will be downloaded if not downloaded before

**Reply by Korkman** [reply]
Attention: this modifies the "MySQL 5" driver for all connections, which is probably unintended! There is a dedicated "MySQL 8" driver which should be selected as [described here](https://github.com/dbeaver/dbeaver/discussions/10641#discussioncomment-229198)

**Comment by rajeshisnepali** [comment]
The steps are good though. However, there is still an error. 

![2020-12-10_16-10](https://user-images.githubusercontent.com/23610013/101760006-68411080-3b02-11eb-9ca3-35f81fe2fcd2.png)
![2020-12-10_16-10_1](https://user-images.githubusercontent.com/23610013/101760002-670fe380-3b02-11eb-9e56-91d5d97f3360.png)

But, If I create a new connection with this **Mysql** driver, it works well.
![image](https://user-images.githubusercontent.com/23610013/101760221-b6eeaa80-3b02-11eb-8140-6415ab357f72.png)

**Comment by kseniiaguzeeva** [comment]
You can use the advice above, but something can go wrong, so I do not recommend it.
To change the driver for the existed connection:
1. Click the connection in DB Navigator
2. Select Database>Change connection driver and select another one
![Screenshot 2020-12-14 at 18 33 27](https://user-images.githubusercontent.com/51405061/102100844-f424a680-3e3a-11eb-9b94-5503ea4a5e19.png)

**Comment by rajeshisnepali** [comment]
> 2\. Select Database>Change connection driver and select another one

It didn't work either.

**Comment by SubOptimal** [comment, marked as answer]
@rajeshisnepali Maybe some of the hints given here could help in your case.
https://community.atlassian.com/t5/Confluence-questions/MySQL-Public-Key-Retrieval-is-not-allowed/qaq-p/778956

This should not be an advice to disable encryption in general. It's rather to be seen as a way to check if the updated connection will work as such. After that please refer to the documentation how to setup the server/client to use an encrypted connection.

**Comment by kseniiaguzeeva** [comment]
@rajeshisnepali Did the last advice help you?

**Comment by rajeshisnepali** [comment]
@kseniiaguzeeva Yes, Of course. 👌

**Comment by xak2000** [comment]
The comment that is marked as answer does not actually answer the question "How to change DB Driver in an existing connection".
The [other comment](https://github.com/orgs/dbeaver/discussions/10641#discussioncomment-229198) should be marked as answer. It directly answers the question.
